src/content-helper/editor-sidebar/smart-linking/component.tsx

```diff
@@ -351,6 +351,28 @@ export const SmartLinkingPanel = ( {
 			return true;
 		} );
 
+		// Exclude Smart Links text and URLs that are already in the post, to avoid duplicates.
+		links = links.filter( ( link ) => {
+			return ! smartLinks.some( ( sl ) => {
+				if ( sl.href === link.href ) {
+					// eslint-disable-next-line no-console
+					console.warn( `PCH Smart Linking: Skipping duplicate link: ${ link.href }` );
+					return true;
+				}
+				if ( sl.text === link.text ) {
+					// If the offset is the same, we want to keep the link, so it can replace the old smart link.
+					if ( sl.offset === link.offset ) {
+						// TODO: Flag smart link as updated.
+						return false;
+					}
+					// eslint-disable-next-line no-console
+					console.warn( `PCH Smart Linking: Skipping duplicate link text: ${ link.text }` );
+					return true;
+				}
+				return false;
+			} );
+		} );
```

Comment:
Thanks for the suggestion, not needed in this case. However, out of curiosity, using `some` instead of `find` isn't better performance wise? Or is it the same?